src/data_base.py

```diff
@@ -1,55 +1,71 @@
-        self.file_path = file_path
-        self.db = sqlite3.connect(self.file_path)
-        self.db.execute('''CREATE TABLE IF NOT EXISTS photos_list 
-                        (id VARCHAR, 
-                        posted BOOL, 
-                        UNIQUE(id) )''')
+    def __init__(self, table_name: str = "photos_list"):
+        self.table_name = table_name
+        parse.uses_netloc.append("postgres")
+        url = parse.urlparse(get_env("HEROKU_DATABASE_URL"))
+        self.connection = psycopg2.connect(
+            database=url.path[1:],
+            user=url.username,
+            password=url.password,
+            host=url.hostname,
+            port=url.port
+        )
+        self.cursor = self.connection.cursor()
+        with self.connection as connection:
+            connection.cursor().execute(sql.SQL('''CREATE TABLE IF NOT EXISTS {}
+                                                    (id VARCHAR,
+                                                    posted BOOL,
+                                                    UNIQUE(id) )''').format(sql.Identifier(self.table_name)))
 
     def __enter__(self):
         return self
 
-    def __exit__(self):
-        self.db.close()
-
-    def add_photo(self, id_photo):
-        self.db.execute("INSERT OR IGNORE INTO photos_list(id, posted) VALUES(?,0)", (id_photo,))
-        self.db.commit()
+    def __exit__(self, *args, **kwargs):
+        self.cursor.close()
+        self.connection.close()
 
-    def unposted_photos(self) -> List[Photo]:
-        a = self.db.execute("SELECT id FROM photos_list WHERE posted = 0")
-        return [row[0] for row in a]
-
-    # def posted_photos(self):
-    #     a = self.db.execute("SELECT id FROM photos_list WHERE posted = 1")
-    #     return [row[0] for row in a]
+    def add_photo(self, id_photo: str):
+        with self.connection as connection:
+            try:
+                connection.cursor().execute(
+                    sql.SQL("INSERT INTO {}(id, posted) VALUES(%s, 'FALSE')").format(sql.Identifier(self.table_name)),
+                    (id_photo,))
+            except psycopg2.IntegrityError:
+                pass
 
     def post_photo(self, post_id: str):
-        self.db.execute("UPDATE photos_list SET posted = 1 WHERE id = ?", (post_id,))
-        self.db.commit()
+        with self.connection as connection:
+            connection.cursor().execute(
+                sql.SQL("UPDATE {} SET posted = 'TRUE' WHERE id = %s").format(sql.Identifier(self.table_name)),
+                (post_id,))
 
-    def delete_photo_from_twitter(self, post_id: str):
-        self.db.execute("UPDATE photos_list SET posted = 0 WHERE id = ?", (post_id,))
-        self.db.commit()
+    def unposted_photos(self) -> List[Photo]:
+        self.cursor.execute(sql.SQL("SELECT id FROM {} WHERE posted = 'FALSE'").format(sql.Identifier(self.table_name)))
+        return [row[0] for row in self.cursor.fetchall()]
```

Comment:
I think you can return generator here, without creating a list

Reply:
random item could not be chosen from generator